pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -887,31 +887,48 @@ func (oe *operationExecutor) generateVerifyControllerAttachedVolumeFunc(
 				volumeToMount.Pod.UID)
 		}
 
+		// This will write to node status that we're about to start using the volume.
+		// If we don't need the given volume, there might be a delay of upto 10s
+		// before we proclaim that we don't need it, because we *don't* synchronously
+		// delete the same volume from the node status map.
+		// TODO: consider fixing this post 1.3.
+		addVolumeNodeErr := actualStateOfWorld.MarkVolumeAsAttached(
+			volumeToMount.VolumeSpec, nodeName, volumeToMount.DevicePath)
+
+		if addVolumeNodeErr != nil {
+
+			actualStateOfWorld.MarkVolumeAsDetached(volumeToMount.VolumeName, nodeName)
```

Comment:
I'm just removing from the list so we don't leak if the pipeline is cancelled by pod delete. It feels like the safe thing to do. Maybe we should defer the asw <-> dsw swap to post 1.3? This might make things a little slower.